docs/development.rst

```diff
@@ -5,16 +5,111 @@
 Development
 ===========
 
-If you want to try out the development version of ``tskit``, or add features or
-documentation improvements, please read the following. If you think there is anything
-missing, please open an `issue <http://github.com/tskit-dev/tskit/issues>`_ or
-`pull request <http://github.com/tskit-dev/tskit/pulls>`_ on GitHub!
+If you would like to add some features to ``tskit``, this
+documentation should help you get set up and contributing.
+Please help us to improve the documentation by either
+opening an `issue <http://github.com/tskit-dev/tskit/issues>`_ or
+`pull request <http://github.com/tskit-dev/tskit/pulls>`_ if you
+see any problems.
 
-.. _sec_development_quickstart:
+.. _sec_development_structure:
 
-**********
-Quickstart
-**********
+*****************
+Project structure
+*****************
+
+Tskit is a multi-language project, which is reflected in the directory
+structure:
+
+- The ``python`` directory contains the Python library and command line interface,
+  which is what most contributors are likely to be interested in. Please
+  see the :ref:`sec_development_python` section for details. The
+  low-level :ref:`sec_development_python_c` interface is also defined here.
+
+- The ``c`` directory contains the high-performance C library code. Please
+  see the :ref:`sec_development_c` for details on how to contribute.
+
+- The ``docs`` directory contains the source for this documentation,
+  which covers both the Python and C APIs. Please see the :ref:`sec_development_documentation`
+  for details.
+
+The remaining files in the root directory of the project are for
+controlling :ref:`sec_development_continuous_integration` providers
+and other administrative purposes.
+
+Please see the :ref:`sec_development_best_practices` section for
+an overview of how to contribute a new feature to ``tskit``.
+
+.. _sec_development_getting_started:
+
+***************
+Getting started
+***************
+
+.. _sec_development_getting_started_requirements:
+
+------------
+Requirements
+------------
+
+To develop the Python code you will need a working C compiler and a
+development installation of Python (>= 3.6). On Debian/Ubuntu we can install these
+with::
+
+    $ sudo apt install python3-dev build-essential doxygen
+
+Python packages required for development are listed in ``python/requirements/development.txt``.
+These can be installed using ``pip``::
+
+    $ python3 -m pip install -r python/requirements/development.txt
+
+You may wish isolate your development environment using a `virtualenv
+<https://docs.python-guide.org/dev/virtualenvs/>`_.
+
+For OSX and Windows users we recommending using
```

Comment:
Sub-section for the OSX/WIndows only stuff? Maybe a box? The last line about c lib deps is easily skipped over otherwise

Reply:
Moved this note up.